lib/validate/net.go

```diff
@@ -22,6 +22,25 @@ import (
 	"github.com/gravitational/trace"
 )
 
+// NetworkOverlap verifies that ipAddr is not in the range of the subnetCIDR
+func NetworkOverlap(ipAddr string, subnetCIDR string, errMsg string) error {
+	_, subNet, err := net.ParseCIDR(subnetCIDR)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	ip := net.ParseIP(ipAddr)
+	if ip == nil {
+		return trace.BadParameter("Invalid IP address (%v).", ipAddr)
```

Comment:
Nit: Usually, we don't capitalize/dot error messages for better embedding ([error strings](https://github.com/golang/go/wiki/CodeReviewComments#error-strings)). An exception would be when the error represents a full message that will be displayed to a user.

```go
return trace.BadParameter("invalid IP address (%v)", ipAddr)
```